pkg/util/conditions/condition_test.go

```diff
@@ -0,0 +1,265 @@
+	role := "master"
+	objectName := "cluster"
+	version := "unknown"
+
+	kubeclock = &clock.FakeClock{}
+	var transitionTime metav1.Time = metav1.Time{Time: kubeclock.Now()}
+
+	for _, tt := range []struct {
+		name      string
+		aroclient aroclient.Interface
+		input     operatorv1.OperatorCondition
+
+		expected arov1alpha1.ClusterStatus
+		wantErr  error
+	}{
+		{
+			name: "noop",
+			aroclient: arofake.NewSimpleClientset(&arov1alpha1.Cluster{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: objectName,
+				},
+			}),
+			expected: arov1alpha1.ClusterStatus{
+				OperatorVersion: version,
+				Conditions:      []operatorv1.OperatorCondition{},
+			},
+		},
+		{
+			name: "clean on version change",
+			aroclient: arofake.NewSimpleClientset(&arov1alpha1.Cluster{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: objectName,
+				},
+				Status: arov1alpha1.ClusterStatus{
+					Conditions: []operatorv1.OperatorCondition{
+						{
+							Type:   arov1alpha1.InternetReachableFromMaster.String(),
+							Status: operatorv1.ConditionFalse,
+						},
+					},
+					OperatorVersion: "?",
+				},
+			}),
+			expected: arov1alpha1.ClusterStatus{
+				Conditions: []operatorv1.OperatorCondition{
+					{
+						Type:   arov1alpha1.InternetReachableFromMaster.String(),
+						Status: operatorv1.ConditionFalse,
+					},
+				},
+				OperatorVersion: version,
+			},
+		},
+		{
+			name: "noop with condition",
+			aroclient: arofake.NewSimpleClientset(&arov1alpha1.Cluster{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: objectName,
+				},
+				Status: arov1alpha1.ClusterStatus{
+					Conditions: []operatorv1.OperatorCondition{
+						{
+							Type:   arov1alpha1.InternetReachableFromMaster.String(),
+							Status: operatorv1.ConditionFalse,
+						},
+					},
+				},
+			}),
+			expected: arov1alpha1.ClusterStatus{
+				Conditions: []operatorv1.OperatorCondition{
+					{
+						Type:   arov1alpha1.InternetReachableFromMaster.String(),
+						Status: operatorv1.ConditionFalse,
+					},
+				},
+				OperatorVersion: version,
+			},
+		},
+		{
+			name: "change with condition",
+			aroclient: arofake.NewSimpleClientset(&arov1alpha1.Cluster{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: objectName,
+				},
+				Status: arov1alpha1.ClusterStatus{
+					Conditions: []operatorv1.OperatorCondition{
+						{
+							Type:   arov1alpha1.InternetReachableFromMaster.String(),
+							Status: operatorv1.ConditionFalse,
+						},
+					},
+				},
+			}),
+			input: operatorv1.OperatorCondition{
+				Type:   arov1alpha1.InternetReachableFromMaster.String(),
+				Status: operatorv1.ConditionTrue,
+			},
+			expected: arov1alpha1.ClusterStatus{
+				Conditions: []operatorv1.OperatorCondition{
+					{
+						Type:               arov1alpha1.InternetReachableFromMaster.String(),
+						Status:             operatorv1.ConditionTrue,
+						LastTransitionTime: transitionTime,
+					},
+				},
+				OperatorVersion: version,
+			},
+		},
+		{
+			name: "preserve with condition",
```

Comment:
Preserve what? Maybe something like this reflects the test case better?

```suggestion
			name: "update one of the existing conditions",
```